qep-345-line-symbollayer-blankareas.md

```diff
@@ -0,0 +1,117 @@
+
+![nonfixed.gif](images/qep345/nonfixed.gif)
+
+This option will be false by default which is the actual behavior.
+
+Enabling this option will imply that we disable feature clipping for the given line symbol layer. The flag `Qgis::SymbolLayerFlag::DisableFeatureClipping` would be added to current flags if this option is set.
+
+When set, all the geometry would be drawn entirely and not clipped. This would have performance implication. A tooltip would inform user about this.
+
+### The blank areas
+
+These areas would be defined as start and end distances from the first point of the line or polygon. It would be possible to define the unit of those distances.
+
+A new property would be added on QgsSymbolLayer [ones](https://github.com/qgis/QGIS/blob/d01be3798eb05c74a61629042f7189ad9c594b1e/src/core/symbology/qgssymbollayer.h#L147) and it would be possible to explicity set a data defined property for all `QgsTemplatedLineSymbolLayerBase` symbol layers (which marker line and hash line symbol layer inherits from). It would be possible then to define those blank areas either:
+- in a static way, 
+- with a specific data field,
+- an auxiliary data storage field,
+- an expression.
+
+When used with a field, this information would then be stored per feature, per layer, per symbol layer.
+
+A new widget to edit those blank areas would be inserted within marker line symbol layer widget.
+
+![blankareas_widget.png](images/qep345/blankareas_widget.png)
+
+#### QgsMapToolDefineBlankArea
+
+A new map tool `QgsMapToolDefineBlankArea` will allow to define these blank areas. 
+
+The new map tool will be accessible through the symbology editor (![mActionRemoveAllFromOverview.svg](images/qep345/mActionRemoveAllFromOverview.svg) in the screenshot above, even though a new icon will be proposed), and only here (because you need to choose the symbol layer you are going to add the blank areas).
+
+Button to activate this map tool would be disable if the symbology editor is not docked, because:
+- The properties window is modal and need to be closed before starting to add blank lines
+- User need some direct visual feedback when defining those blank areas
+
+A tooltip will explain why when it's disabled.
+
+This tool will allow to:
+- Click in the neighborhood of the line to start creating an area, click again to finish editing the zone like on the video above.
+- Select an area, press `Del` key to remove it
+- Drag the start/end of an already existing area and move it along the line to reduce/enlarge it
+- Drag the zone entirely and move it along the line
+
+the map tool defined zone would be set on `QgsTemplatedLineSymbolLayerBase` symbol layers as data defined properties. If no data defined property has been set, a window would ask user if he wants to generate an auxiliary storage layer and which field he wants to use to join.
```

Comment:
Please use similar logic to the label tools to determine whether a field is available to store the values in -- there is nice logic there which handles expression based data defined values, eg `case when @map_scale > 20000 then "label_x_zoomed_out" else "label_x_zoomed_in" end` and extract that actual target field from that.

Reply:
I fail to find track of what you're mentioning. Where does that take place exactly?

Reply:
Done aa15c1d